core/src/main/java/hudson/model/Queue.java

```diff
@@ -2131,6 +2132,16 @@ default long getEstimatedDuration() {
             return Executables.getParentOf(this).getEstimatedDuration();
         }
 
+        /**
+         * Handles cases such as {@code PlaceholderExecutable} for Pipeline node steps.
```

Comment:
Did you test this case to make sure it works?

Reply:
This is actually moved code from `Functions#getConsoleUrl` but I doublechecked to make sure.